src/commands/join.cpp

```diff
@@ -1,12 +1,12 @@
-/* Copyright 2024 <faboussa>************************************************* */
+/* ************************************************************************** */
 /*                                                                            */
 /*                                                        :::      ::::::::   */
 /*   join.cpp                                           :+:      :+:    :+:   */
 /*                                                    +:+ +:+         +:+     */
-/*   By: faboussa <faboussa@student.42lyon.fr>      +#+  +:+       +#+        */
+/*   By: fanny <faboussa@student.42lyon.fr>         +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/10/17 11:50:56 by faboussa          #+#    #+#             */
-/*   Updated: 2024/11/19 14:35:36 by faboussa         ###   ########.fr       */
+/*   Updated: 2024/11/24 13:54:27 by fanny            ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
```

Comment:
We should use `receiveMessage()` function of Client class instead of `send()`. In `receiveMessage()` function`, log print is included, so it helps us to debug or find potential issues.

```cpp
void Server::sendJoinMessageToClient(int fd, const std::string &nick,
                                     const std::string &channelName,
                                     const Client &client) {
  std::string joinMessage = ":" + nick + " JOIN :#" + channelName + "\r\n";
  if (send(fd, joinMessage.c_str(), joinMessage.length(), 0) == -1) {
    printLog(ERROR_LOG, SYSTEM, "JOIN: send failed");
    throw std::runtime_error(RUNTIME_ERROR);
  }
  if (_channels[channelName].getTopic().topic.empty())
    send331Notopic(client, _channels[channelName]);
  else
    send332Topic(client, _channels[channelName]);
}
```